InternalDocs/README.md

```diff
@@ -33,9 +33,11 @@ Runtime Objects
 Program Execution
 ---
 
-- [The Interpreter](interpreter.md)
+- [The Basic Interpreter](interpreter.md)
 
-- [Adaptive Instruction Families](adaptive.md)
+- [The Specializing Interpreter](adaptive.md)
+
+- [The Tier 2 Interpreter (coming soon)](tier2.md)
```

Comment:
@markshannon, should we call this the "JIT interpreter"? That matches the general shift in naming we've been using, and matches the configure option `--enable-experimental-jit=interpreter`.

(It's beyond the scope of this PR, obviously, but we should probably go through and clean up all occurrences of `r"[Tt][Ii][Ee][Rr][_ ]?(2|[Tt][Ww][Oo])"` in the codebase.)

```suggestion
- [The JIT Interpreter (coming soon)](jit_interpreter.md)
```

Reply:
I was thinking of writing two documents - "the tier 2 interpreter" describes the optimiser and executors mechanism. 
Then another doc about the "experimental jit interpreter" which has the stencils business. 

I think it will be more accessible broken up like this.

Reply:
I'd call the thing that runs with `--enable-experimental-jit` the "JIT compiler" and the thing that runs with `--enable-experimental-jit=interpreter` either the "tier two interpreter" or "jit interpreter". The former doesn't really run your code in an interpreter, so I think calling it that could be confusing.

But naming is hard, and the term "interpreter" is super overloaded already. Eh.

Reply:
The code under `tier2_dispatch` is not an interpreter?

Reply:
It is. But that code is [`#ifdef`'d out](https://github.com/python/cpython/blob/d24a22e9b6377797c3b602945347096fbe065670/Python/ceval.c#L1038-L1040) unless the build is configured with `--enable-experimental-jit=interpreter`. Instead of compiling to machine code, it will interpret the micro-ops, which is useful for people debugging issues in tier 2 that aren't directly related to machine code (no user should use this interpreter, it's very slow).

When the build is configured with bare `--enable-experimental-jit`, `_Py_JIT` is defined, which makes sure that executors are [compiled](https://github.com/python/cpython/blob/d24a22e9b6377797c3b602945347096fbe065670/Python/optimizer.c#L1202-L1213), and it [redefines a couple of macros](https://github.com/python/cpython/blob/d24a22e9b6377797c3b602945347096fbe065670/Python/ceval_macros.h#L381-L404) to jump into the compiled code instead of using the tier 2 interpreter.

More info on the difference here: https://docs.python.org/3/using/configure.html#cmdoption-enable-experimental-jit

Reply:
This is internals documentation, it's supposed to help people who want to work on the code get to grips with the how thing are implemented. Explaining Tier 2 first (as a uops interpreter) and then layering the jit compilation on top of that seems more didactic to me than describing it from a user's POV.

Reply:
Yeah, I think that approach makes sense too.

Reply:
I changed my mind about this - it's all in one doc now.